amaranth_soc/csr/field.py

```diff
@@ -0,0 +1,188 @@
+from amaranth import *
+
+from .reg import Field
+
+
+__all__ = ["R", "W", "RW", "RW1C", "RW1S"]
+
+
+class R(Field):
+    __doc__ = Field._doc_template.format(
+    description="""
+    A read-only field.
+    """.strip(),
+    parameters="",
+    attributes="""
+    r_data : Signal(shape)
+        Read data. Drives the :attr:`~FieldPort.r_data` signal of ``port``.
+    """.strip())
+
+    def __init__(self, shape):
+        super().__init__(shape, access="r")
+        self.r_data = Signal(shape)
+
+    def elaborate(self, platform):
+        m = Module()
+        m.d.comb += self.port.r_data.eq(self.r_data)
+        return m
+
+
+class W(Field):
+    __doc__ = Field._doc_template.format(
+    description="""
+    A write-only field.
+    """.strip(),
+    parameters="",
+    attributes="""
+    w_data : Signal(shape)
+        Write data. Driven by the :attr:`~FieldPort.w_data` signal of ``port``.
+    """.strip())
+
+    def __init__(self, shape):
```

Comment:
This should take `reset` like `field.RW`.

One hypothetical utility is a scenario where `reset=-1` could be used to let the gateware know that the host hasn't initialized a particular value yet, if 0 is a valid value for the register.

Reply:
I don't think this would do anything, as the line in `elaborate` will override any reset value specified.

Reply:
I see, I overlooked the fact that the field doesn't have any storage.

Then adding a reset value is indeed meaningless.